app/publisher/publisher.py

```diff
@@ -0,0 +1,37 @@
+import google
+from google.cloud.pubsub_v1 import PublisherClient
+from google.cloud.pubsub_v1.futures import Future
+from structlog import get_logger
+
+from app.publisher.publication_failed import PublicationFailed
+
+logger = get_logger(__name__)
+
+
+class PubSub(PublisherClient):
+    def __init__(self, topic_id):
+        super().__init__()
+
+        _, project_id = google.auth.default()
+        self.topic_path = super().topic_path(project_id, topic_id)
+
+    def _publish(self, message):
+        logger.info("publishing message")
+        publish_future: Future = super().publish(self.topic_path, message)
+
+        return publish_future
+
+    def publish_and_resolve_message(self, message: bytes):
```

Comment:
This method should be just called `publish` as the resolve is irrelevant to the calling method.

Reply:
Updated but doing this and the ABC change meant I had to remove the `PublisherClient` inheritance due to miss matching method signatures.